lib/src/installer/completion_configuration.dart

```diff
@@ -56,18 +60,33 @@ class CompletionConfiguration {
     }
 
     return CompletionConfiguration._(
-      uninstalls: _jsonDecodeUninstalls(decodedJson),
+      uninstalls: _jsonDecodeUninstalls(
+        decodedJson,
+        jsonKey: CompletionConfiguration._uninstallsJsonKey,
+      ),
+      installs: _jsonDecodeUninstalls(
+        decodedJson,
+        jsonKey: CompletionConfiguration._installsJsonKey,
+      ),
     );
   }
 
   /// The JSON key for the [uninstalls] field.
   static const String _uninstallsJsonKey = 'uninstalls';
 
+  /// The JSON key for the [installs] field.
+  static const String _installsJsonKey = 'installs';
+
   /// Stores those commands that have been manually uninstalled by the user.
   ///
   /// Uninstalls are specific to a given [SystemShell].
   final Uninstalls uninstalls;
 
+  /// Stores those commands that have completion installed.
+  ///
+  /// Installed commands are specific to a given [SystemShell].
+  final Uninstalls installs;
```

Comment:
I don't get we have to keep a map of all commands. If we have this lib installed on a cli, other installations should not concern this routine. In other words, this algorithm will never use more the one element in these collections, especially "installs"

Reply:
Why is this "installs" collection necessary?